registry/storage/blobwriter.go

```diff
@@ -89,11 +91,12 @@ func (bw *blobWriter) Cancel(ctx context.Context) error {
 		return err
 	}
 
+	bw.Close()
```

Comment:
Maybe at least log it since it should not be occurring but may be useful to know why a canceled blob exists.